src/auth/authclient.ts

```diff
@@ -407,9 +407,14 @@ export abstract class AuthClient
       headers.set('x-goog-user-project', this.quotaProjectId);
     }
 
-    const trustBoundaryHeader = this.getTrustBoundaryHeader();
-    if (trustBoundaryHeader !== null) {
-      headers.set('x-allowed-locations', trustBoundaryHeader);
+    if (this.trustBoundaryEnabled && this.trustBoundary) {
```

Comment:
I noticed you changed this to include `this.trustBoundaryEnabled` but I don't see an added unit test. Is this covered by a unit test?